apio/resources/boards.json

```diff
@@ -119,6 +119,20 @@
       "pid": "2100"
     }
   },
+  "TinyFPGA-BX": {
+    "name": "TinyFPGA BX",
+    "fpga": "iCE40-LP8K-CM81",
+    "programmer": {
+      "type": "tinyprog"
+    },
+    "usb": {
+      "vid": "1d50",
+      "pid": "6130"
+    },
+    "tinyprog": {
+      "desc": "TinyFPGA BX"
```

Comment:
This information is not used. Now a system to detect the FTDI description is implemented (`ftdi.desc`) using the internals of `apio system --lsftdi` to detect and verify the boards.
```
$ apio system --lsftdi
Number of FTDI devices found: 1
Checking device: 0
Manufacturer: Mareldem, Description: IceZUM Alhambra v1.1 - B01-020
```

We could implement something similar using `tinyprog` to verify the board description. Could you paste the output of `tinyprog --list`?

Reply:
Looking the output of `tinyprog -l` I guess the description is next to the port name `TinyFPGA BX 1.0.0`. I could execute this command if the boards has a `tinyprog` field and then use regex to extract a list of active board descriptions.

```
tinyprog -l

    TinyProg CLI
    ------------
    Using device id 1d50:6130
    Only one board with active bootloader, using it.
    Boards with active bootloaders:

        COM14: TinyFPGA BX 1.0.0
            UUID: e518a627-5626-4f92-91f5-07ed26503bb9
            FPGA: ice40lp8k-cm81
```

Since the description is also a regular expression it should be changed to `"desc": "TinyFPGA BX.*".

Reply:
I am planning on adding more information to the bootloader so that it can provide all the information icestudio and apio need. I may change the "bootmeta/update" key to a more generate URL that contains this type of information.

Reply:
OK, so the steps to check the `tinyprog.desc` verification (after the serial verification) would be:

1. Check if the board configuration has a `tinyprog.desc` field
2. Use `tinyprog -m` to obtain the JSON metadata (`M`)
3. Verify that the detected port (`p`) matches the metadata (`M[p]`) : `tinyprog.desc` in `M[p].boardmeta.name`

Now if more than one boards are detected and pass all the checks Apio uses the first one. In order to select a specific board the `--serial-port` argument could be used.

Are you OK with the plan/implementation of the new check for the `tinyprog` programmable boards?